CMakeLists.txt

```diff
@@ -1,105 +1,86 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.5)
+CMake_minimum_required(VERSION 3.0)
 
-project(booredis)
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake")
-string(TOUPPER ${PROJECT_NAME} PROJECT_NAME_UPPER)
+project(booredis VERSION 0.2.0)
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
 include(GNUInstallDirs)
 
-set(LIB_MAJOR_VERSION "0")
-set(LIB_MINOR_VERSION "1")
-set(LIB_PATCH_VERSION "7")
-
-set(LIB_VERSION_STRING "${LIB_MAJOR_VERSION}.${LIB_MINOR_VERSION}.${LIB_PATCH_VERSION}")
-
-
 # compile in release with debug info mode by default
-IF(NOT CMAKE_BUILD_TYPE)
-  SET(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING
-      "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel."
-      FORCE)
-ENDIF(NOT CMAKE_BUILD_TYPE)
-
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING
+        "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel."
+        FORCE)
+endif()
 
-OPTION(BOOREDIS_BUILD_SHARED "Build booredis as a shared library" ON)
-OPTION(BOOREDIS_BUILD_CXX11 "Build booredis with C++11" ON)
-OPTION(BOOREDIS_BUILD_EXAMPLES "Build booredis examples" ON)
-OPTION(BOOREDIS_BUILD_UNITTESTS "Automatically build unit tests" ON)
+option(BUILD_EXAMPLES "Build booredis examples" ON)
 
+# Subdirs
+add_subdirectory(src)
 
-# Select to build shared or static library
-IF(BOOREDIS_BUILD_SHARED)
```

Comment:
https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html

Reply:
>  This variable is often added to projects as an OPTION so that each user of a project can decide if they want to build the project using shared or static libraries.

So, maybe create such option then?

Reply:
also see https://github.com/google/fruit/issues/44 for similar case and issues caused by this global option when the project is included as a subdirectory to the build tree.